Implementado create_account na currencyApi

GCastilho opened the pull request:
Geral:
-- A página de cadastro agora requisita uma account nano da CurrencyApi ao salvar o usuário no banco de dados
-- Currencies agora tem uma propriedade 'ip', que indica o IP da máquina em que seu módulo externo está rodando
-- A propriedade que identifica o nome da currency nos módulos individuais agora se chama 'name' e não 'currency'
-- A propriedade 'name' de uma currency na CurrencyApi agora é colocado automaticamente pelo index da currencies, não sendo necessário ter uma propriedade para isso

Módulos individuais:
-- Implementado validador de nano accounts ao salvar um usuário no database
-- Porta da NANO configurada para 8090

Listener:
-- Refatorado módulo listener para separar funções acessíveis pelas APIs externas através do listener das funções internas da CurrencyApi
-- As funções acessíveis externamente devem ficar no pseudo-módulo 'functions', com cada função sendo um sub-módulo diferente
-- As funções do módulo 'commons' agora só são acessíveis internamente, não estando disponíveis para serem acessíveis pelo listener

Bugs:
-- Corrigido problema de o cache do require do módulo das currencies estar sendo modificado (globalmente) na instanciação da CurrencyApi

Testes:
-- Adicionado mock (manual) na CurrencyApi para não chamar o módulo externo ao requisitar uma account NANO nova nos testes
-- Adicionado testes para garantir que as funções do listener são acessíveis e funções individuais ou da commons não
-- Adicionado testes no listener para comando inexistente e currency inexistente

gabr1gus committed:
prototipo de servidor para api da Nano

--criaçao de funcoes baseadas nos comandos rcp
--criaçao de um servidor que ira lidar com comandos post e get direcionados a api da Nano
--adicao de um client para o websocket da nano, que verifica as operacoes realizadas na rede

gabr1gus committed:
minor change

--response do get é enviado como parametro para ser tratado dentro da funcao que é chamada dentro de rcp

GCastilho committed:
Reorganização de arquivos

-- Movido módulo para uma pasta 'external_modules'
-- Criado npm module próprio para o módulo nano
-- Adicionado código guia para utilização do axios para ler stream do servidor

gabr1gus committed:
criado lista de accounts no mongo

--collections de accounts é dropada e recriada na inicializaçao do modulo
--vars trocado pra consts
--; removidos
--terminou!!1! corrigido para terminou!!!!

gabr1gus committed:
small tweaks and reorganization

--adicionado comentarios
--reorganizado algumas partes de código
--adicionado uma chamada ao websocket no codigo principal

GCastilho committed:
Código refatorado

-- Refatorada a estrutura dos módulos e do código dentro dos módulos sem alterar a funcionalidade deles
-- Corrigido package.json e instalado módulos ausentes
-- Adicionado env para o mongodb, porta do servidor e url do servidor principal

GCastilho committed:
Minor fix

-- Pequeno refatoramento da Promise chain do index
-- Removido função 'exit', que que agora o exit só é chamado em um único lugar do código

gabr1gus committed:
createAccount agora retorna uma promisse

--minor changes
--removed trash

GCastilho committed:
Merge branch 'master' into create-account-backend

GCastilho committed:
Adicionada função create_account() na common da currencyApi

-- Adicionado proteção para funções privadas não poderem ser acessadas pelo listener (função privadas são as que começam com '_')
-- Adicionada função 'create_account' na common da currencyApi
-- Porta da NANO configurada para 8090
-- Minor fixes

GCastilho committed:
Trocado propriedade de currency para name

-- A propriedade que identifica o nome da currency nos mósulos individuais agora se chama 'name' e não 'currency'
-- Modificado teste e corrigido todo o código para essa mudança

GCastilho committed:
O nome da currency agora é adicionado automaticamente

-- A propriedade 'name' de uma currency na CurrencyApi agora é colocado automaticamente pelo index da currencies, não sendo necessário ter uma propriedade para isso

GCastilho committed:
Currencies agora tem uma propriedade 'ip'

-- Currencies agora tem uma propriedade 'ip', que indica o IP da máquina em que seu módulo externo está rodando

GCastilho committed:
Refatorado módulo listener

-- Refatorado módulo listener para separar funções acessíveis pelas APIs externas através do listener das funções internas da currencyApi
-- As funções acessíveis externamente devem ficar no pseudo-módulo 'functions', com cada função sendo um sub-módulo diferente
-- As funções do módulo 'commons' agora só são acessíveis internamente, não estando disponíveis para serem acessíveis pelo listener
-- O refatoramento não afetou funcionalidades externas, a única exceção é que não é mais necessário passar a currencyApi como argumento para o listener ao carregá-lo

GCastilho committed:
Adicionado testes para garantir que funções individuais ou da commons não estão acessíveis pelo listener

-- Corrigido problema de o cache do require do módulo das currencies estar sendo modificado (globalmente) na instanciação da currencyApi
-- Adicionado testes para garantir que as funções do listener são acessíveis e funções individuais ou da commons não
-- Adicionado testes no listener para comando inexistente e currency inxistente
-- Minor fixes

GCastilho committed:
Implementado validador de accounts NANO

-- A página de cadastro agora requisita uma account nano da currencyApi ao salvar o usuário no banco de dados
-- Implementado validador de nano accounts ao salvar um usuário no database
-- create_account() now throw an Error when the request fails

GCastilho committed:
Corrigido testes estarem invocando módulo externo para pedir NANO accounts ao fazer o cadastro

-- Corrigido testes estarem invocando módulo externo para pedir NANO accounts ao fazer o cadastro
-- Adicionado mock (manual) na currencyApi para não chamar o módulo externo ao requisitar uma account NANO nova

joaojvds reviewed (approved):
Eu vou dar um approve porque no final os tests passaram, mas de uma olhada para ver se os resultados dos tests não estão interferindo com os outros tests

GCastilho committed:
Revisado alguns testes

-- Revisado e corrigido a possibilidade de alguns testes estarem interferindo em outros e fazendo-os falhar
-- Um dos testes de cadastro ainda PODE falhar por race condition
-- Alguns outros fixes menores

GCastilho committed:
Merge branch 'master' into create-account-backend
